File: plugins/capawesome-capacitor-wifi/android/src/main/java/io/capawesome/capacitorjs/plugins/wifi/classes/results/GetIpAddressResult.java
package io.capawesome.capacitorjs.plugins.wifi.classes.results;

import android.net.wifi.WifiInfo;
import androidx.annotation.NonNull;
import com.getcapacitor.JSObject;
import io.capawesome.capacitorjs.plugins.wifi.WifiHelper;
import io.capawesome.capacitorjs.plugins.wifi.interfaces.Result;

public class GetIpAddressResult implements Result {

    private int ipAddress;

    public GetIpAddressResult(@NonNull WifiInfo wifiInfo) {
        this.ipAddress = wifiInfo.getIpAddress();
    }

    public JSObject toJSObject() {
        String ipAddressString = WifiHelper.convertIpAddressIntToString(ipAddress);

        JSObject result = new JSObject();
        result.put("address", ipAddressString);
        return result;
    }
}

File: plugins/capawesome-capacitor-wifi/ios/Plugin/Classes/Results/GetIpAddressResult.swift
import Foundation
import CoreBluetooth
import Capacitor

@objc public class GetIpAddressResult: NSObject, Result {
    private let ipAddress: String

    init(ipAddress: String) {
        self.ipAddress = ipAddress
    }

    public func toJSObject() -> AnyObject {
        var result = JSObject()
        result["address"] = ipAddress
        return result as AnyObject
    }
}
